packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -657,10 +663,14 @@ export default createRule<Options, MessageId>({
       AssignmentExpression: checkAssignmentExpression,
       BinaryExpression: checkIfBinaryExpressionIsNecessaryConditional,
       CallExpression: checkCallExpression,
-      ConditionalExpression: (node): void => checkNode(node.test),
+      ConditionalExpression: (node): void => {
+        checkNode(node.test);
+      },
```

Comment:
TBH I personally don't think that multilining arrows is necessarily clearer.
Is there a reason the stylistic set is pushing for this style?

Reply:
Yup it is! +1 to the visibility @rubiesonthesky 😄.

I wonder if it should be moved to the `strict-type-checked` config instead... https://github.com/typescript-eslint/typescript-eslint/discussions/7130#discussioncomment-6285562